src/pudl/analysis/ferc1_eia.py

```diff
@@ -674,6 +674,40 @@ def prep_train_connections(
         "plant_part",
         "ownership_dupe",
     ]
+    # Read in one_to_many csv and join corresponding plant_match_ferc1 parts to FERC IDs
+    one_to_many = (
+        (
+            pd.read_csv(
+                importlib.resources.path(
+                    "pudl.package_data.glue", "ferc1_eia_one_to_many.csv"
+                ),
+            )
+            .pipe(pudl.helpers.cleanstrings_snake, ["record_id_eia"])
+            .drop_duplicates(subset=["record_id_ferc1", "record_id_eia"])
+        )
+        .merge(ppe[["record_id_eia", "ferc1_generator_agg_id"]])
+        .dropna(subset=["ferc1_generator_agg_id"])
+        .drop(["record_id_eia"], axis=1)
+        .merge(
+            ppe.loc[
+                ppe.plant_part == "plant_match_ferc1",
+                ["record_id_eia", "ferc1_generator_agg_id"],
+            ],
+            on="ferc1_generator_agg_id",
+        )
+        .drop(["ferc1_generator_agg_id"], axis=1)
```

Comment:
although the outcome would be exactly the same, I think `drop_duplicates` before the second merge of the `ppe` would make it feel more clear to me. bc if i understand correctly this initial merge is an attempt to find the one `ferc1_generator_agg_id` that really matters for each many grouping.

+ possibly adding merge `on="record_id_eia", how="left"` into the first merge

Reply:
> One question about this: do all of the non-generator plant parts have these ferc1_generator_agg_ids? Or will it basically always be the record_id_eia's that show up in the one:many CSV that have the ferc1_generator_agg_id? Do you need to run the match_to_single_plant_part to convert everything into generators first if the generators are the only ones with this ferc1_generator_agg_id?

@cmgosnell Not 100% sure I understand the question, but the original 1:m IDs in ferc_eia_one_to_many.csv are a mix of non-generator plant parts and generator plant parts. Then I use the `match_to_single_plant_part()` function to identify all the component generators, de-duplicate, and label these generators _only_ with the `ferc1_generator_agg_id`. These are the records that get aggregated to form the faked plant parts (so, plant parts -> generators -> single plant part). Is your question whether the non-generator aggregate IDs that are the original sources for the 1:m also labelled with the group ID?

In this step I'm taking the original one to many CSV, using the FERC ID to identify what plant part got matched in the end to the FERC ID, and replacing the original match with the faked part (plant part - FERC ID -> faked plant part - FERC ID).

Reply:
> although the outcome would be exactly the same, I think `drop_duplicates` before the second merge of the `ppe` would make it feel more clear to me. bc if i understand correctly this initial merge is an attempt to find the one `ferc1_generator_agg_id` that really matters for each many grouping.
> 
>     * possibly adding merge `on="record_id_eia", how="left"` into the first merge

The second merge matches each sub-component ID to its corresponding match part. Because it's a m:1 match, when we drop the ID column we wind up with duplicates of the faked plant part IDs (one for each sub-component of the new plant part). So the drop_duplicates() should stay where it is, but I've added some validations in the merges to make expectations explicit.